### ISSUE TITLE
Dynamic variables support for ImportControl check (property expansion, maybe)

### ISSUE THREAD

**Issue by merzbird** [checkstyle/checkstyle#6791]
Problem: cannot find a way to create generic import-control config to allow import  for some subpackages only from the same modules - really useful for multimodule project with the same module structures and rules.

From the current available options to resolve this issue the most suitable one is probably property expansion.
Currently it is not possible to use property expansion functionality for ImportControl configuration, as it is always must be a separate config file linked by root config, for instance:
```
<module name="Checker">
    <module name="TreeWalker">
        <module name="ImportControl">
            <property name="file" value="config/import-control.xml"/>
        </module>
    </module>
</module>
```
While it is possible to declare variable properties using `${prop_name}` syntax in the snippet above, one cannot do the same with `import-control.xml` file.

With this support it must be allowed to write something like this.
```
<import-control pkg="com.example.${module_name}">
    <subpackage name=".*\.core.*" regex="true">
        <allow pkg="com\.example\.${module_name}\.api*" regex="true"/>
    </subpackage>
</import-control>
```

Any ideas of how to resolve the problem with different approaches are really appreciated.

**Comment by romani** [comment]
Try to follow issue template to share all required for us details to help you. Issue template is provided on good reason to avoid ping pong of messages to get all details from author of issue.
Imagine your case is a bug.

We have property resolving , see https://checkstyle.org/config.html#Properties .
I can not help you more as I have no idea how exactly to execute validation, by plugin or ?

**Comment by merzbird** [comment]
@romani Thanks for the reply. I updated my post with details as described in the template. Hope it will help.

**Comment by romani** [comment]
I agree to allow users to use variable in config, there might be some conflict between regexp values and variables substitution.  But I doubt that users reasonably can use `....${name}` as regexp. So at least they will not define variable `name` so, it will stay as-is without value substitution.

@rnveach and @pbludov , please review and approve issue.

**Comment by rnveach** [comment]
I am fine with this.

**Comment by romani** [comment]
I placed approve label, but @pbludov , can raise red flag if found some issues.

@merzbird , the quickest way to make it, is to send PR, your are welcome with PR.

**Comment by pbludov** [comment]
I'm OK with this approach.

As for collisions between regular expressions and variable names, the `$` symbol can occur in regular expressions, but usually only at the end. There shouldn't be a problem.
